packages/database/src/core/RepoManager.ts

```diff
@@ -80,7 +83,11 @@ export class RepoManager {
    * @return {!Database}
    */
   databaseFromApp(app: FirebaseApp, url?: string): Database {
-    const dbUrl: string = url || app.options[DATABASE_URL_OPTION];
+    const dbEmulatorHost = process.env[FIREBASE_DATABASE_EMULATOR_HOST_VAR];
+    const dbUrl: string =
+      url || app.options[DATABASE_URL_OPTION] || dbEmulatorUrl
+        ? `http://${dbEmulatorHost}`
+        : undefined;
```

Comment:
1) Isn't app.options[DATABASE_URL_OPTION] almost always going to be set?  And so I'm not sure what we gain by falling back to the environment variable?  It seems like we'd want the environment variable to *override* the options. But maybe I'm misunderstanding the use case?
2) Offhand I don't know the operator precedence between `||` and the ternary operator (`... ? ... : ...`) and so this is hard to read.  I would advocate adding parentheses or splitting this up.

Reply:
1. One use case is when a developer calls `initializeApp` with an empty options object, but has the environment variable set. In this case we want to default to pointing at the emulator.
2. I'll split it up.

Reply:
Why would the developer call initializeApp() with an empty options object though?  That won't ever work (except if you're using this environment variable), right?  And if you are modifying you code to work with the emulator, you might as well just call initializeApp() and pass the emulator URL for your databaseURL, right?

Reply:
Yep, that's true. I think maybe the variable should override app.options, but not the passed url parameter. The reason is that if the firebase admin sdk already sees the variable, it will pass in the appropriate url as an argument to this package (in which case we want to prefer the argument). However, if the developer is using the package directly, they may want to use this variable to override what is in `databaseURL`.